tests/FINDOLOGIC/Tests/Objects/XmlResponseTest.php

```diff
@@ -0,0 +1,346 @@
+    {
+        $expectedName = 'ordernumber';
+        $expectedValue = '019111105-37900';
+
+        $response = $this->getRealResponseData();
+
+        foreach ($response->getProducts() as $product) {
+            foreach ($product->getProperties() as $propertyName => $propertyValue) {
+                $this->assertEquals($expectedName, $propertyName);
+                $this->assertEquals($expectedValue, $propertyValue);
+            }
+        }
+    }
+
+    public function testResponseWillReturnFiltersAsExpected()
+    {
+        $expectedFilterDisplays = ['Preis', 'Farbe', 'Material', 'Hersteller', 'Kategorie'];
+        $expectedFilterTypes = ['range-slider', 'color', 'select', 'select', 'select'];
+        $expectedFilterNames = ['price', 'Farbe', 'Material', 'vendor', 'cat'];
+        $expectedFilterSelects = ['single', 'multiselect', 'multiple', 'multiple', 'single'];
+        $expectedFilterSelectedItems = [0, 1, 0, 0, 0];
+        $expectedFilterAmount = 5;
+
+        $response = $this->getRealResponseData();
+
+        $count = 0;
+        foreach ($response->getFilters() as $filter) {
+            $this->assertEquals($expectedFilterDisplays[$count], $filter->getDisplay());
+            $this->assertEquals($expectedFilterTypes[$count], $filter->getType());
+            $this->assertEquals($expectedFilterNames[$count], $filter->getName());
+            $this->assertEquals($expectedFilterSelects[$count], $filter->getSelect());
+            $this->assertEquals($expectedFilterSelectedItems[$count], $filter->getSelectedItems());
+            $this->assertEquals($expectedFilterAmount, $response->getFilterAmount());
+            $count++;
+        }
+    }
+
+    public function testResponseWillReturnAttributesAsExpected()
+    {
+        $expectedSelectedRange = ['min' => 0.39, 'max' => 2239.10];
+        // There is no difference between selected and total if the result was not filtered by a range-slider.
+        $expectedTotalRange = ['min' => 0.39, 'max' => 2239.10];
+        $expectedStepSize = 0.1;
+        $expectedUnit = '€';
+
+        $response = $this->getRealResponseData();
+
+        foreach ($response->getFilters() as $filter) {
+            // Only range-slider does have attributes.
+            if ($filter->getType() === 'range-slider') {
+                $this->assertEquals($expectedSelectedRange, $filter->getAttributes()->getSelectedRange());
+                $this->assertEquals($expectedTotalRange, $filter->getAttributes()->getTotalRange());
+                $this->assertEquals($expectedStepSize, $filter->getAttributes()->getStepSize());
+                $this->assertEquals($expectedUnit, $filter->getAttributes()->getUnit());
+            } else {
+                // All other filter types do not have attributes.
+                $this->assertEquals(null, $filter->getAttributes());
+            }
+        }
+    }
+
+    public function testResponseWillReturnItemsAsExpected()
```

Comment:
For the Smart Suggest you have an individual test method per field, here you have one test method for all fields. This is both fine, but please consider having similar test structures.

Reply:
I was just lazy here... Now I read the code and I am like: "If one thing does not work as expected, only this test will fail and I will manually have to check what exactly makes this test failing".